olmo/train.py

```diff
@@ -727,6 +782,21 @@ def train_batch(self, batch: Dict[str, Any]) -> Tuple[torch.Tensor, Optional[tor
 
         ce_batch_loss = torch.tensor(0.0, device=self.device)
         z_batch_loss = None if not self.cfg.softmax_auxiliary_loss else torch.tensor(0.0, device=self.device)
+        lb_batch_loss = (
+            None if self.model.config.block_type != BlockType.moe else torch.tensor(0.0, device=self.device)
+        )
+        moe_z_batch_loss = (
```

Comment:
Why is MoE zloss different from regular zloss? What happens if you enable both?

Reply:
It's explained in this paper: https://arxiv.org/pdf/2202.08906 - essentially the MoE one is applied to routers
You can activate both if you want though I haven't tried it